File: backend/src/main/java/com/ornitologo/backend/models/Localizacao.java
package com.ornitologo.backend.model;

import javax.persistence.Embeddable;

@Embeddable
public class Localizacao {
    private String lat;
    private String longt;
    private String descricao;

    public String getLat() {
        return lat;
    }

    public void setLat(String lat) {
        this.lat = lat;
    }

    public String getLongt() {
        return longt;
    }

    public void setLongt(String longt) {
        this.longt = longt;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

}

File: frontend/src/models/Localizacao.ts
import { ILocalizacao } from "../interfaces/ILocalizacao";

export class Localizacao implements ILocalizacao{
    lat: string;
    longt: string;
    descricao: string;

    constructor(body: ILocalizacao){
        this.lat = body.lat;
        this.longt = body.longt;
        this.descricao = body.descricao? body.descricao: "";
    }
}